pkg/logs/internal/launchers/file/launcher.go

```diff
@@ -60,10 +61,20 @@ type Launcher struct {
 }
 
 // NewLauncher returns a new launcher.
-func NewLauncher(tailingLimit int, tailerSleepDuration time.Duration, validatePodContainerID bool, scanPeriod time.Duration) *Launcher {
+func NewLauncher(tailingLimit int, tailerSleepDuration time.Duration, validatePodContainerID bool, scanPeriod time.Duration, wildcardMode string) *Launcher {
+	var prov *fileprovider.FileProvider
+	if wildcardMode == "prioritizeNewest" {
+		prov = fileprovider.NewFileProvider(tailingLimit, fileprovider.WildcardMtime, fileprovider.GlobalSelection)
+	} else if wildcardMode == "simpleAlpha" {
+		prov = fileprovider.NewFileProvider(tailingLimit, fileprovider.WildcardReverseLexicographical, fileprovider.GreedySelection)
+	} else {
+		log.Warnf("Unknown wildcard mode specified: %q, defaulting to 'simpleAlpha' strategy.\n", wildcardMode)
+		prov = fileprovider.NewFileProvider(tailingLimit, fileprovider.WildcardReverseLexicographical, fileprovider.GreedySelection)
+	}
```

Comment:
Would it make sense to call `NewFileProvider` once, and use a this block instead to only pick appropriate wildcard mode to pass to the constructor? Perhaps `switch` would be appropriate here too.